app-libs/stf/src/stf_sgx_sidechain.rs

```diff
@@ -0,0 +1,76 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+use crate::{helpers::get_storage_value, Stf};
+use codec::Encode;
+use itp_sgx_externalities::SgxExternalitiesTrait;
+use itp_stf_interface::system_pallet::SystemPalletSidechainInterface;
+use itp_storage::storage_value_key;
+use itp_types::{BlockHash, SidechainBlockNumber, SidechainTimestamp as Timestamp};
+use its_primitives::traits::Block as SidechainBlockTrait;
+use std::prelude::v1::*;
+
+impl<Call, Getter, State, SidechainBlock> SystemPalletSidechainInterface<State, SidechainBlock>
```

Comment:
Introduced a (not yet) used Sidechain System trait & its implementaiton. Not sure if we will need that, but I think it would make sense to access the state at one place. Be it sidechain or getter and calls.

Reply:
I am not sure too, but I think I would like an issue that writes a sidechain pallet that we use in the sgx-runtime only, that implements all these storage values so that we do not need to do the low-level sp-io calls here. This pallet should live in the sgx-runtime crate as it is not intended to live on the parentchain.

Reply:
Proposing the follow up issue:
> implement a sidechain pallet, that allows sidechain specific runtime calls. When implement this, also evaluate if the whole SidechainDB struct is still needed, or if everything can be directly accessed via the Stf.

Reply:
#1036